### PR TITLE
Add binary encoding for Distinct

### PR DESCRIPTION
**Pull request by tvaron3**
Extends the existing opt-in Cosmos binary JSON query-page support to cross-partition `DISTINCT` queries.

## What changed

- Normalizes binary feed envelopes at the shared `Distinct`/`SkipTake` splitting boundary and preserves the negotiated encoding for synthesized pages.
- Preserves DISTINCT equality when text and binary pages represent the same integral number differently.
- Covers mixed text/binary `ORDER BY` → `DISTINCT` → `SkipTake`, continuation tokens, fuzz cases, and live partition splits.

This PR does not change query negotiation, query-plan requests, or read-feed behavior.

### CODE REVIEW

File: sdk/cosmos/azure_data_cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
...
 - Added cross-partition `DISTINCT` query support. `SELECT DISTINCT` now deduplicates structurally equal values across every physical partition and page, rather than failing as an unsupported query feature. A `DISTINCT` query of the exact form `SELECT DISTINCT VALUE <path> … ORDER BY <same path>` (for example `SELECT DISTINCT VALUE c.city FROM c ORDER BY c.city`) is resumable from a continuation token; every other shape — including a list projection such as `SELECT DISTINCT c.city …` and any multi-column `ORDER BY` — is not, and requesting a token for it returns an error explaining how to rewrite the query. ([#5026](https://github.com/Azure/azure-sdk-for-rust/pull/5026))
 - Added an SDK-generated `x-ms-client-id` header that remains stable for each `CosmosClient`. ([#4844](https://github.com/Azure/azure-sdk-for-rust/pull/4844))
+- Extended opt-in Cosmos binary JSON encoding to cross-partition `DISTINCT` query pages. ([#5070](https://github.com/Azure/azure-sdk-for-rust/pull/5070))
```
[the remark this answers is not in the document]
Comment (tvaron3) [reply sdk/cosmos/azure_data_cosmos/CHANGELOG.md]: Removed the incorrect `Fixes #5185` reference. That issue tracks follow-up binary vector-query support and should remain open.

File: sdk/cosmos/azure_data_cosmos_driver/src/driver/dataflow/query_response.rs [hunk 2]
```diff
@@ -302,7 +307,7 @@ pub(crate) fn parse_envelope_page(
 ) -> crate::error::Result<Vec<EnvelopeRow>> {
     let bytes = match body {
         ResponseBody::NoPayload => return Ok(Vec::new()),
-        ResponseBody::Bytes(b) => b.clone(),
+        ResponseBody::Bytes(b) => normalize_page_body(b)?,
```
[the remark this answers is not in the document]
Comment (tvaron3) [reply sdk/cosmos/azure_data_cosmos_driver/src/driver/dataflow/query_response.rs]: Removed the redundant normalization from the `ResponseBody::Bytes` match arm. ORDER BY pages now pass through the shared normalization point exactly once before parsing.

**Review by ananth7592 — APPROVED** [review]
LGTM. 
Few non blocking comments
The test coverage here looks strong, especially the text/binary parity, cross-encoding continuation, mixed-page, numeric-equivalence, and live partition-split scenarios.

Since this change builds on the binary codec and negotiation behavior already reviewed in the recently merged binary-encoding work, I’m treating that foundation as established rather than asking this PR to revalidate the wire format independently.


I have two non-blocking questions about regression coverage:

- Could we add one full-pipeline test where malformed binary arrives after DISTINCT has consumed earlier state, confirming that further execution and continuation creation remain poisoned?

- Can we confirm that the gated live partition-split suite runs regularly outside normal PR validation?

I don’t see a correctness issue in the current implementation; these would primarily strengthen regression protection.

**Comment by tvaron3** [comment]
Thanks, @ananth7592.

- Good catch on the malformed-binary case. We separately test malformed input inside `Distinct` and plan poisoning after an output transcode failure, but not the exact composed sequence. I’ll add a full-pipeline test with one valid DISTINCT page followed by malformed binary, verifying that the second execution fails, continuation creation is refused, and subsequent execution remains poisoned.
- The split suite runs regularly outside normal PR validation. `sdk/cosmos/ci.yml` enables live tests for scheduled and `- weekly` definitions, and `live-platform-matrix.json` includes the `Session Split` leg. I also verified that the August 30 scheduled `rust - cosmos - weekly` run executed `Test windows_stable_SessionSplit` successfully.

File: sdk/cosmos/azure_data_cosmos/tests/emulator_tests/cosmos_query.rs [hunk 3]
```diff
@@ -29,6 +29,50 @@ fn collect_matching_items(
...
+        "query returned a different number of items"
+    );
+
+    let mut remaining = actual_items;
+    for expected in expected_items {
+        let Some(index) = remaining.iter().position(|actual| actual == &expected) else {
+            panic!("query result is missing {expected:?}; unmatched actual items: {remaining:?}");
+        };
+        remaining.swap_remove(index);
+    }
```
Comment (analogrelay) [inline sdk/cosmos/azure_data_cosmos/tests/emulator_tests/cosmos_query.rs:64]: You need to test that `remaining` is empty at the end of the loop. Otherwise `actual_items` could contain additional items without failing the check.

File: sdk/cosmos/azure_data_cosmos_driver/src/lib.rs [hunk 4]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT License.
 
 #![doc = include_str!("../README.md")]
+#![recursion_limit = "256"]
```
Comment (analogrelay) [inline sdk/cosmos/azure_data_cosmos_driver/src/lib.rs]: This updates the compiler's recursion limit when evaluating things like macros and auto-dereference. What triggered the old limit to be exceeded?

**Review by analogrelay — CHANGES_REQUESTED** [review]
Small test fix needed

File: sdk/cosmos/azure_data_cosmos/tests/emulator_tests/cosmos_query.rs (hunk 3 above)
Comment (tvaron3) [reply sdk/cosmos/azure_data_cosmos/tests/emulator_tests/cosmos_query.rs:64]: Good point. I added a check that `remaining` is empty.

File: sdk/cosmos/azure_data_cosmos_driver/src/lib.rs (hunk 4 above)
Comment (tvaron3) [reply sdk/cosmos/azure_data_cosmos_driver/src/lib.rs]: This was triggered by the binary codec cargo-fuzz harness added in #4977. With nightly-2026-09-08, sanitizer instrumentation, and `-D warnings`, rustc exceeded the default depth while proving `Send` through the container-recreation async chain (`try_recover_recreated_container` through `ContainerCache`, `AsyncCache`, and `AsyncLazy`). I reproduced the failure locally without the attribute and verified the same build succeeds with it. I also scoped the higher limit to `cfg(fuzzing)` so normal builds retain Rust's default, and registered that cfg with the build script.

Comment (analogrelay) [reply sdk/cosmos/azure_data_cosmos_driver/src/lib.rs]: The `cfg(fuzzing)` change is appreciated

**Review by analogrelay — APPROVED** [review]

**Review by ananth7592 — APPROVED** [review]
